### PR TITLE
Add breakpad

### PR DESCRIPTION
**Pull request by gsocnikos**
This is my first pull request for my breakpad branch :-)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -154,9 +154,13 @@ if (APPLE)
 else (APPLE)
    if (MINGW)
       # -mno-ms-bitfields see #22048
-      set(CMAKE_CXX_FLAGS_DEBUG   "-std=c++11 -mno-ms-bitfields -g")
-      set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
-      set(CMAKE_EXE_LINKER_FLAGS "-Wl,--large-address-aware")
+      set(CMAKE_CXX_FLAGS_DEBUG          "-g -std=c++11 -mno-ms-bitfields")
+	  set(CMAKE_C_FLAGS_DEBUG            "-g -mno-ms-bitfields")
```
Comment (lasconic) [inline CMakeLists.txt]: Please fix indentation here.

File: CMakeLists.txt [hunk 2]
```diff
@@ -556,7 +560,7 @@ ADD_CUSTOM_TARGET(mops2 DEPENDS ${PCH})
 subdirs(
       mscore  awl bww2mxml share midi audiofile fluid mstyle libmscore synthesizer
       effects thirdparty/rtf2html thirdparty/diff thirdparty/beatroot
-      thirdparty/qzip thirdparty/kQOAuth
+      thirdparty/qzip thirdparty/kQOAuth thirdparty/breakpad
```
Comment (lasconic) [inline CMakeLists.txt]: it would probably be good to add an option for breakpad, just like OMR, ZERBERUS etc... below.

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+#  - %APPDATA%\npm-cache             # npm cache
+# build platform, i.e. x86, x64, Any CPU. This setting is optional.
+platform:
+  - x86
+# build Configuration, i.e. Debug, Release, etc.
+configuration: Release
+# scripts to run before build
+before_build:
+  - IF NOT EXIST dependencies.7z (
+    start " " /wait "C:\MinGW\msys\1.0\bin\wget" --no-check-certificate "https://drive.google.com/uc?export=download&id=0BxjayMZiuupOSXF6a0ZDUG5rNDg" -O dependencies.7z &&
```
Comment (lasconic) [inline appveyor.yml]: Let's move this file to a more durable place... 
Use this URL http://utils.musescore.org.s3.amazonaws.com/musescore_dependencies_win32.7z

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  - set PATH=C:\Qt\5.8\mingw53_32\bin;C:\Qt\Tools\mingw530_32\bin;C:\Program Files (x86)\CMake\bin;C:\Program Files\7-Zip;C:\ccache\bin
+  - set CCACHE_DIR=C:\ccache\cache
+  - set /p MSversion=<mscore\revision.h
+# scripts to run *after* solution is built and *before* automatic packaging occurs (web apps, NuGet packages, Azure Cloud Services)
+#before_package:
+# scripts to run after build
+after_build:
+  - ccache.exe -s
+  - cd C:\MuseScore\win32install\bin
+  - start " " /wait C:\MinGW\msys\1.0\bin\wget --no-check-certificate "https://github.com/gsocnikos/musescore_crash/raw/master/cv2pdb/cv2pdb.exe"
```
Comment (lasconic) [inline appveyor.yml]: let's zip or 7zip those and store them on S3 too. You can then unzip them and use them.

Comment (lasconic) [inline appveyor.yml]: Add a README mentioning where you found the code, and how you compiled it.

**Comment by lasconic** [comment]
As you can see here https://travis-ci.org/musescore/MuseScore/builds/252496505?utm_source=github_status&utm_medium=notification the build failed on Linux.

File: appveyor.yml (hunk 4 above)
Comment (lasconic) [inline appveyor.yml]: You can add the README here https://github.com/gsocnikos/musescore_crash/tree/master/cv2pdb and I will create the zip and store it on s3. Tell me here or on IRC when it's ready

Comment (lasconic) [inline appveyor.yml]: If there are other references to your github, they need to go.

File: build/mingw32.cmake [hunk 5]
```diff
@@ -4,8 +4,8 @@
 
 set (CMAKE_SYSTEM_NAME "Windows")
 
-set (CROSS /home/ws/mingw)
```
Comment (lasconic) [inline build/mingw32.cmake]: You don't need to change this file. It's only used for crosscompiling the windows version on linux. Please revert the changes.

File: mscore/CMakeLists.txt [hunk 6]
```diff
@@ -436,20 +436,31 @@ if (MINGW)
...
+          )
+   endif(CMAKE_BUILD_TYPE MATCHES "RELWITHDEBINFO")
 
    target_link_libraries(mscore
       portaudio
       portmidi
       winmm
       mscore_freetype
       z
+      breakpad
```
Comment (lasconic) [inline mscore/CMakeLists.txt]: Like above, would be good to add an option

File: mscore/musescore.cpp [hunk 7]
```diff
@@ -5708,8 +5712,17 @@ using namespace Ms;
 //   main
 //---------------------------------------------------------
 
+int buggyFunc() {
```
Comment (lasconic) [inline mscore/musescore.cpp]: please delete this.

File: mscore/musescore.cpp [hunk 8]
```diff
@@ -5708,8 +5712,17 @@ using namespace Ms;
...
+    delete reinterpret_cast<QString*>(0xFEE1DEAD);
+    return 0;
+}
+
 int main(int argc, char* av[])
       {
+
```
Comment (lasconic) [inline mscore/musescore.cpp]: delete these lines.

File: mscore/musescore.cpp [hunk 9]
```diff
@@ -5793,6 +5806,11 @@ int main(int argc, char* av[])
 
       parser.process(QCoreApplication::arguments());
 
+      //Instantiate breakpad instanse currently will produce the minidumps under the c:\Users\username directory
```
Comment (lasconic) [inline mscore/musescore.cpp]: instan**c**e

File: mscore/musescore.cpp [hunk 10]
```diff
@@ -5793,6 +5806,11 @@ int main(int argc, char* av[])
...
+#ifdef _WIN32
+      Breakpad::CrashHandler::instance()->Init(QStandardPaths::writableLocation(QStandardPaths::HomeLocation));
```
Comment (lasconic) [inline mscore/musescore.cpp]: We need to find another place to write the crashes. Suggestions?

File: mscore/musescore.cpp [hunk 11]
```diff
@@ -6257,6 +6275,9 @@ int main(int argc, char* av[])
       if (settings.value("mixerVisible", false).toBool())
             mscore->showMixer(true);
 
+      //TESTING BREAKPAD
```
Comment (lasconic) [inline mscore/musescore.cpp]: Please remove this.

File: CMakeLists.txt [hunk 12]
```diff
@@ -590,6 +595,12 @@ if (OMR)
       subdirs (omr thirdparty/poppler)
 endif (OMR)
 
+if (BREAKPAD)
+    subdirs (thirdparty/breakpad)
+    include_directories(${PROJECT_SOURCE_DIR}/thirdparty/breakpad)
+
```
Comment (lasconic) [inline CMakeLists.txt]: one blank line

File: CMakeLists.txt [hunk 13]
```diff
@@ -624,6 +635,9 @@ if (OMR)
    include_directories(${PROJECT_SOURCE_DIR}/thirdparty/poppler/poppler)
 endif (OMR)
 
+
```
Comment (lasconic) [inline CMakeLists.txt]: more blank lines

File: mscore/musescore.cpp [hunk 14]
```diff
@@ -123,6 +123,12 @@ extern Ms::Synthesizer* createAeolus();
 extern Ms::Synthesizer* createZerberus();
 #endif
 
+#ifdef BREAKPAD
+#ifdef _WIN32
```
Comment (lasconic) [inline mscore/musescore.cpp]: Just don't define breakpad in Cmake if not on WIN32 and remove this test.

**Comment by lasconic** [comment]
Please fix the other problems and check that the build can pass on linux. It would be cool if it's done by tomorrow standup meeting.

File: mscore/musescore.cpp [hunk 15]
```diff
@@ -5710,6 +5714,10 @@ using namespace Ms;
 
 int main(int argc, char* av[])
       {
+
+
+
+
```
Comment (Jojo-Schmitz) [inline mscore/musescore.cpp]: excess newlines

File: appveyor.yml [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+#  allow_failures:
+#    - platform: x86
+#      configuration: Release
+# build cache to preserve files/folders between builds
+cache:
+  - dependencies.7z
+  - C:\Qt\Tools\mingw530_32\lib
+  - C:\Qt\Tools\mingw530_32\i686-w64-mingw32
+  - C:\Program Files (x86)\Jack
+  - C:\Qt\Tools\mingw530_32\libexec\gcc\i686-w64-mingw32\5.3.0\cc1plus.exe
```
Comment (lasconic) [inline appveyor.yml]: this line is not needed.

File: appveyor.yml [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+  - x86
+# build Configuration, i.e. Debug, Release, etc.
+configuration: Release
+# scripts to run before build
+before_build:
+  - IF NOT EXIST dependencies.7z (
+    start " " /wait "C:\MinGW\msys\1.0\bin\wget" --no-check-certificate "http://utils.musescore.org.s3.amazonaws.com/musescore_dependencies_win32.7z" -O dependencies.7z &&
+    start " " /wait "7z" x -y dependencies.7z > nul &&
+    cd dependencies &&
+    xcopy i686-w64-mingw32 "C:\Qt\Tools\mingw530_32\i686-w64-mingw32" /E &&
```
Comment (lasconic) [inline appveyor.yml]: this is probably not needed.

Comment (lasconic) [inline appveyor.yml]: it needs to be check.

File: appveyor.yml [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+    xcopy Jack "C:\Program Files (x86)\Jack" /E /I &&
+    xcopy ccache "C:\ccache" /E /I
+    )
+  # copy /y cc1plus.exe "C:\Qt\Tools\mingw530_32\libexec\gcc\i686-w64-mingw32\5.3.0\cc1plus.exe"
+  
+  - cd C:\MuseScore
+  - set PATH=C:\Qt\5.8\mingw53_32\bin;C:\Qt\Tools\mingw530_32\bin;%PATH%
+  - mingw32-make -f Makefile.mingw revision
+  # CMake refuses to generate MinGW Makefiles if sh.exe is in the Path
+  - set PATH=C:\Qt\5.8\mingw53_32\bin;C:\Qt\Tools\mingw530_32\bin;C:\Program Files (x86)\CMake\bin;C:\Program Files\7-Zip;C:\ccache\bin
```
Comment (lasconic) [inline appveyor.yml]: You override PATH here. That's why you don't have curl in the PATH.

File: appveyor.yml (hunk 17 above)
Comment (AntonioBL) [inline appveyor.yml]: It copies the header files of ogg, vorbis, lame, portaudio, etc. into the compiler "include" folder, so I think it is needed.
It was not needed (and indeed it was canceled) the replacement of cc1plus.exe, which was a patched version of the executable included in Qt tools to allow for pre-compiled headers generation, but now it is no more necessary thanks to the new gcc version. (therefore it can also be removed from the 7zip archive)

Comment (lasconic) [inline appveyor.yml]: @AntonioBL thank you for checking this PR! And great to see that your effort on appveyor will pay out !

Comment (lasconic) [inline appveyor.yml]: I removed cc1plus.exe from the 7zip package. Could you try to build with appveyor again?

File: CMakeLists.txt [hunk 19]
```diff
@@ -154,9 +161,13 @@ if (APPLE)
 else (APPLE)
    if (MINGW)
       # -mno-ms-bitfields see #22048
-      set(CMAKE_CXX_FLAGS_DEBUG   "-std=c++11 -mno-ms-bitfields -g")
-      set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
-      set(CMAKE_EXE_LINKER_FLAGS "-Wl,--large-address-aware")
+      set(CMAKE_CXX_FLAGS_DEBUG          "-std=c++11 -mno-ms-bitfields -g")
+      set(CMAKE_C_FLAGS_DEBUG            "-mno-ms-bitfields -g")
```
Comment (lasconic) [inline CMakeLists.txt]: The C_FLAGS are really needed? only for breakpad? if yes, could you put them around a BREAKPAD condition?

File: CMakeLists.txt (hunk 13 above)
Comment (lasconic) [inline CMakeLists.txt]: still blank lines here. Please remove them.

File: mscore/musescore.cpp [hunk 20]
```diff
@@ -5793,6 +5798,10 @@ int main(int argc, char* av[])
 
       parser.process(QCoreApplication::arguments());
 
+      //Initiate breakpad instance currently will produce the minidumps under the c:\Users\username directory
+#ifdef BREAKPAD
+      Breakpad::CrashHandler::instance()->Init(QStandardPaths::writableLocation(QStandardPaths::HomeLocation));
```
Comment (lasconic) [inline mscore/musescore.cpp]: I would use `dataPath`+ / crash_reports 
See `MuseScore::writeSessionFile`

File: mscore/musescore.cpp [hunk 21]
```diff
@@ -5710,6 +5714,7 @@ using namespace Ms;
 
 int main(int argc, char* av[])
       {
+
```
Comment (lasconic) [inline mscore/musescore.cpp]: Delete this line please

File: mscore/musescore.cpp [hunk 22]
```diff
@@ -5946,6 +5951,17 @@ int main(int argc, char* av[])
       // if not already there:
       QDir().mkpath(dataPath + "/plugins");
 
+      //Initiate breakpad instance currently will produce the minidumps under the c:\Users\username directory
+#ifdef BREAKPAD
+      QString breakpad_path = dataPath+"/crash_reports";
+      if (! QDir(breakpad_path).exists()){
```
Comment (lasconic) [inline mscore/musescore.cpp]: checking the existence is not necessary http://doc.qt.io/qt-5/qdir.html#mkpath

**Comment by AntonioBL** [comment]
I am not an expert in this field, therefore do not take for granted what I say, but what about using QProcess::startDetached?
http://doc.qt.io/qt-5/qprocess.html#startDetached
From what they say in that page, the working directory is inherited from the calling process if that parameter is left empty, and I think it should be MuseScore bin directory, so you could simply put the exe file alongside MuseScore.exe.
(And I also think that QProcess would work also in Linux and Mac, so if in the future the dump report is extended to those platforms as well, an almost platform-independent implementation at this stage could help).

But better waiting for lasconic's comments :-)

**Comment by lasconic** [comment]
I might be wrong, but at this time, we don't have the luxury of using Qt anymore. MuseScore just crashed and we want to fork another process to report the crash.

**Comment by lasconic** [comment]
Btw, another approach would be to have a server running... and instead of forking at crash time, send info to the server. This is the approach of crashpad I believe https://chromium.googlesource.com/crashpad/crashpad/+/master/handler/crashpad_handler.md

**Comment by AntonioBL** [comment]
Actually, I tried to locally compile this branch and using QProcess::startDetached in the if condition instead of CreateProcess, and it worked!
:-)

I simply used:
QProcess::startDetached("crashReporter.exe")
and I put file crashReporter.exe in the win32install/bin folder.
Here is the command line output I received after a crash:
`Debug: BreakpadQt crash (C:\Git_packages\MuseScore\thirdparty\breakpad\crash_handler.cpp:92, bool Breakpad::DumpCallback(const wchar_t*, const wchar_t
*, void*, EXCEPTION_POINTERS*, MDRawAssertionInfo*, bool))
Debug: minidump path: C:/Users/antonio/AppData/Local/MuseScore/MuseScoreDevelopment/crash_reports/e419a49b-5644-4abb-b2b5-d2c020658c4b.dmp
 (C:\Git_packages\MuseScore\thirdparty\breakpad\crash_handler.cpp:101, bool Breakpad::DumpCallback(const wchar_t*, const wchar_t*, void*, EXCEPTION_PO
INTERS*, MDRawAssertionInfo*, bool))
CrashReporter: C:/Users/nickhatz/MuseScore/build.release/thirdparty/breakpad/crashReporter.exeDmppath: C:/Users/antonio/AppData/Local/MuseScore/MuseSc
oreDevelopment/crash_reports/e419a49b-5644-4abb-b2b5-d2c020658c4b.dmpDebug: {"response":"ok","object":"d"}

 (C:\Git_packages\MuseScore\thirdparty\breakpad\crash_handler.cpp:152, bool Breakpad::DumpCallback(const wchar_t*, const wchar_t*, void*, EXCEPTION_PO
INTERS*, MDRawAssertionInfo*, bool))
HELLO!!!`

A couple of observations:
- I had to comment out "||  (code == DBG_PRINTEXCEPTION_WIDE_C)" at line 486 of thirdparty/breakpad/client/windows/handler/exception_handler.cc because it was giving an error during compilation ("missing ) before ;")
- I had to install the Windows SDK in order to compile breakpad, and I had to change the target_link_libraries of line 37 of thirdparty/breakpad/CMakelists.txt since the version I installed (10.0.15063.0) is different from the hardcoded one
- just after the crash, my firewall warns me that MuseScore is trying to establish a connection with 45.33.83.92 TCP, port 6098

If it may help (especially for the compilation problem), my configuration is:
Windows 10 Home
gcc 7.1.0 (i686-posix-dwarf) of MinGW-W64 project
Qt 5.8.0

Ciao,
ABL

**Comment by Jojo-Schmitz** [comment]
That IP adress looks suspicious, as per nslookup it is li1029-92.members.linode.com

**Comment by lasconic** [comment]
45.33.83.92 is musescore.sp.backtrace.io (and apparently they are hosted by linode.com)
https://github.com/musescore/MuseScore/pull/3236/files#diff-d054f4b98d344b7c9967c6e79f184c7aR129


Backtrace.io is the service we might use to gather the information for the crashes. 
If we go with them (it's not free, even for open source), we don't have to setup and maintain a server, and their service would do the desymbolification process, group similar crashes together, possibly integrate with IRC etc... Only a dozen of users would have access to the server.

Alternative is to maintain a server ourself, with open source code to do the crash gathering. Then we can go wild, like mozilla https://crash-stats.mozilla.com/home/product/Firefox but we need the man power to maintain the server and the software :-/

**Comment by gsocnikos** [comment]
Hello Antonio,

the alternative to fork() for windows would be CreateProcess

https://msdn.microsoft.com/en-us/library/windows/desktop/ms682425(v=vs.85).aspx

windows does have a weird way how they implement Operating System operations

Best Regards,
--Nikos

**Comment by gsocnikos** [comment]
I ll try to finish this implementation with breakpads,

The reason why I did not consider crashpad was that they do not support
linux

breakpad the chrash_handler runs on a different thread that's why you need
to launch it on the beggining of main() of mscore

I will finish the dialog window and probably start seeing the crashpad
option

Best Regards,
--Nikos

File: thirdparty/breakpad/mainwindow.h [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+
+class MainWindow : public QMainWindow {
+    Q_OBJECT
+
+public:
+    explicit MainWindow(QWidget *parent = 0);
+    ~MainWindow();
+
+private slots:
+    void on_pushButton_2_clicked();
```
Comment (lasconic) [inline thirdparty/breakpad/mainwindow.h]: Please give these buttons a proper name

File: thirdparty/breakpad/mainwindow.cpp [hunk 24]
```diff
@@ -0,0 +1,334 @@
...
+
+        minidump_path = QCoreApplication::arguments().at(1);
+        metadata_path = QCoreApplication::arguments().at(2);
+
+        QStringList filePathList = minidump_path.split('/');
+        QString minidump_filename = filePathList.at(filePathList.count() - 1);
+
+        qDebug() << "minidump file: " << minidump_path;
+
+        QString url = "https://musescore.sp.backtrace.io:6098/post?format=minidump&token=00268871877ba102d69a23a8e713fff9700acf65999b1f043ec09c5c253b9c03";
```
Comment (lasconic) [inline thirdparty/breakpad/mainwindow.cpp]: Please put this in a define CRASH_SUBMIT_URL

File: thirdparty/breakpad/mainwindow.cpp [hunk 25]
```diff
@@ -0,0 +1,334 @@
...
+
+        QString boundary = "--"
+                    + QString::number(
+                            qrand() * (90000000000) / (RAND_MAX + 1) + 10000000000, 16);
+
+        QNetworkRequest request(url);
+        request.setHeader(QNetworkRequest::ContentTypeHeader,
```
Comment (lasconic) [inline thirdparty/breakpad/mainwindow.cpp]: Add a comment here to explain that the boundary needs to be added in the header too.

File: thirdparty/breakpad/mainwindow.cpp [hunk 26]
```diff
@@ -0,0 +1,334 @@
...
+                    }
+
+            }
+    }
+
+    return pair<string,string>(str1,str2);
+
+}
+
+QMap <QString,QString> read_csv(QString mypath){
```
Comment (lasconic) [inline thirdparty/breakpad/mainwindow.cpp]: this is more than just reading a random CSV. Give it a descriptive name and add some comments before the function.

File: thirdparty/breakpad/mainwindow.cpp [hunk 27]
```diff
@@ -0,0 +1,334 @@
...
+
+    for(i=0;i<path_n;i++){
+        res += mscore_path[i] ;
+    }
+
+    return QString(res.c_str());
+}
+
+
+QString get_musescore_path(){
```
Comment (lasconic) [inline thirdparty/breakpad/mainwindow.cpp]: Add comment before the function

File: thirdparty/breakpad/mainwindow.cpp [hunk 28]
```diff
@@ -0,0 +1,334 @@
...
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
+
+bool launcher(wstring program){
```
Comment (lasconic) [inline thirdparty/breakpad/mainwindow.cpp]: Add comment before the function.

**Comment by lasconic** [comment]
It would be good to:

* Add comments before each function in `mainwindows.cpp`
* MuseScore uses 6 spaces indent and align the closing `}` with previous block. See https://musescore.org/en/developers-handbook/musescore-coding-rules (which unfortunately it's a bit broken right now...) or just check a CPP file in `libmscore` or `mscore` directory
* Since it's your code and not thirdparty, it would be good to move the crashreporter out of `thirdparty/breakpad` and put it in `tools/musescore_crashreporter` for example. `thirdparty/breakpad` is supposed to be easily upgradable.

File: thirdparty/breakpad/mainwindow.cpp [hunk 29]
```diff
@@ -0,0 +1,334 @@
...
+    //CloseHandle( pi.hProcess );
+    //CloseHandle( pi.hThread );
+
+
+    Q_UNUSED(program);
+    return false;
+
+}
+
+QString get_crashreporter_path(){
```
Comment (lasconic) [inline thirdparty/breakpad/mainwindow.cpp]: Maybe you can just use `qApp-> applicationDirPath()` ? That would make this code portable.
See http://doc.qt.io/qt-5/qcoreapplication.html#applicationDirPath

File: thirdparty/breakpad/mainwindow.cpp [hunk 30]
```diff
@@ -0,0 +1,334 @@
...
+    wstring res(mystr.begin(), mystr.end());
+    return res;
+}
+
+string wstr2str(wstring mystr){
+    string res(mystr.begin(), mystr.end());
+    return res;
+}
+
+pair<string,string> line2strings(string line){
```
Comment (lasconic) [inline thirdparty/breakpad/mainwindow.cpp]: Please add comments for the function. Why string and not `QString`?
See QString::split() http://doc.qt.io/qt-5/qstring.html#split

File: CMakeLists.txt [hunk 31]
```diff
@@ -556,7 +564,7 @@ ADD_CUSTOM_TARGET(mops2 DEPENDS ${PCH})
 subdirs(
       mscore  awl bww2mxml share midi audiofile fluid mstyle libmscore synthesizer
       effects thirdparty/rtf2html thirdparty/diff thirdparty/beatroot
-      thirdparty/qzip thirdparty/kQOAuth
+      thirdparty/qzip thirdparty/kQOAuth tools/musescore_crashreporter
```
Comment (lasconic) [inline CMakeLists.txt]: You probably need to put this between if (BREAKPAD)

File: tools/musescore_crashreporter/mainwindow.cpp [hunk 32]
```diff
@@ -0,0 +1,347 @@
...
+
+//-----------------------------------------------------------------------------
+// get_musescore_path() : finds the path of MuseScore.exe
+// MuseScore.exe can be found in the current directroy (for production)
+// or under the mscore folder (during development build)
+//-----------------------------------------------------------------------------
+
+QString get_musescore_path()
+      {
+      QString crashreporter_path = QCoreApplication::applicationDirPath().replace("/", "\\");
```
Comment (lasconic) [inline tools/musescore_crashreporter/mainwindow.cpp]: Are the \\ really needed ? Qt should cope with direct /

File: tools/musescore_crashreporter/mainwindow.cpp [hunk 33]
```diff
@@ -0,0 +1,347 @@
...
+            return res;
+
+      return res_empty;
+      }
+
+//-----------------------------------------------------------------------------
+// str2wstr : convert a string to wstring
+//-----------------------------------------------------------------------------
+
+wstring str2wstr(string mystr)
```
Comment (lasconic) [inline tools/musescore_crashreporter/mainwindow.cpp]: are str2wstr and wstr2str still used ?

File: tools/musescore_crashreporter/mainwindow.cpp [hunk 34]
```diff
@@ -0,0 +1,347 @@
...
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
+
+//-----------------------------------------------------------------------------
+// launcher(program) : executes a file with filepath: program
+// this function is only valid under windows Operating System
+// when the executable is executed the process is send to the background
+//-----------------------------------------------------------------------------
+
+bool launcher(wstring program)
```
Comment (lasconic) [inline tools/musescore_crashreporter/mainwindow.cpp]: This one can probably be made more cross platform. See QProcess::startDetached http://doc.qt.io/qt-5/qprocess.html#startDetached

**Comment by lasconic** [comment]
The code is working on windows. Unfortunately breakpad is becoming obsolete and we don't have a good solution server side to get the crash logs and deal with them. I close this PR for now.
